classes/UpgradeTools/CoreUpgrader/CoreUpgrader17.php

```diff
@@ -112,4 +118,54 @@ protected function upgradeLanguage($lang)
             $cldrUpdate->fetchLocale(\Language::getLocaleByIso($isoCode));
         }
     }
+
+    protected function updateTheme()
+    {
+        parent::updateTheme();
+
+        $this->updateRTLFiles();
+    }
+
+    protected function updateRTLFiles()
+    {
+        if (!$this->container->getUpgradeConfiguration()->shouldUpdateRTLFiles()) {
+            return;
+        }
+        $this->logger->info($this->container->getTranslator()->trans('Upgrade the RTL files.', array(), 'Modules.Autoupgrade.Admin'));
+        $themeAdapter = new ThemeAdapter($this->db, $this->destinationUpgradeVersion);
+
+        $themes = $themeAdapter->getListFromDisk();
+        $this->removeExistingRTLFiles($themes);
+
+        foreach ($themes as $theme) {
+
+            $adaptThemeToTRLLanguages = new AdaptThemeToRTLLanguagesCommand(
+                new ThemeName($theme['name'])
+            );
+
+            /** @var CommandBusInterface $commandBus */
+            $commandBus = $this->container->getModuleAdapter()->getCommandBus();
```

Comment:
It will not work for all 1.7 version, you need to be sure the command bus is available
The AdaptThemeToRTLLanguagesCommand has been added in [1.7.6.0](https://github.com/PrestaShop/PrestaShop/blob/1.7.6.0/src/Core/Domain/Theme/Command/AdaptThemeToRTLLanguagesCommand.php)

Reply:
So I need to verify with `class_exists ` if the class exists before launching my script. If I do this it's okay for you?

Reply:
If you do this, this process will only work if target version is higher than 1.7.6

- if user upgrades from 1.7.3.0 to 1.7.7.5 ✅ 
- if user upgrades from 1.7.7.8 to 1.7.8.2 ✅ 
- if user upgrades from 1.7.3.0 to 1.7.5.1 🟥